docs/routing/introduction.md

```diff
@@ -0,0 +1,37 @@
+# Introduction
+
+Most JavaScript frameworks come with a dedicated routing solution
+like `angular/router` or `vue-router`. They make it possible to
+navigate through different pages of an application without having to
+do a full page refresh on every link click.
+Since we no longer have one monolithic application that handles all routes, instead we have several independent applications, we need to solve an important issue, namely routing.
+
+The following describes how this is handled in the ILC, but first let's clear up the terminology:
+
+ - Hard transitions - Transition in which the application on the page is replaced by another one.
```

Comment:
emphasize the fact that one is when routing is handled via ILC, and another is when routing handled inside the fragment